File: 프로그래머스/1/132267. 콜라 문제/콜라 문제.cs
using System;

public class Solution 
{
    public int solution(int a, int b, int n) 
    {
        int count = 0;
        while(n / a > 0)
        {
            int t1 = n / a;
            int t2 = t1 * a;
            count += t1 * b;
            n -= t2;
            n += t1 * b;
        }
        return count;
    }
}

File: 프로그래머스/1/132267. 콜라 문제/콜라 문제.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

int solution(int a, int b, int n) {
    int sum = 0;
    int m = 0;
    while(n>=a)
    {
        m = (n/a)*b;
        n = n - ((n/a)*a) + (n/a)*b;
        sum += m;
        cout << m;
    }
    return sum;   
}